### PR TITLE
Replace REST API links and references on Learn index with those to RPC

### PR DESCRIPTION
**Pull request by Clay-Mysten**

### CODE REVIEW

File: doc/src/learn/index.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Find the latest updates to these contents in this section:
 
 * [JSON-RPC API](../build/json-rpc.md) - Set up your own local Sui RPC Server and use the Sui JSON-RPC API to interact with a local Sui network.
 * [Narwhal and Tusk for consensus](../learn/architecture/consensus.md) - Learn about Narwhal and Tusk, Sui's high-throughput mempool and consensus engine.
-* [REST API publishing](../build/rest-api.md#post-apipublish) - Instructions exist for publishing Move modules via the Publish endpoint.
+* [REST API publishing](../build/json-rpc.md#sui_publish) - Instructions exist for publishing Move modules via the Publish endpoint.
```
Comment (patrickkuo) [inline doc/src/learn/index.md:19]: Should be `RPC API .....`

**Review by patrickkuo — CHANGES_REQUESTED** [review]

File: doc/src/learn/index.md (hunk 1 above)
Comment (Clay-Mysten) [reply doc/src/learn/index.md:19]: > Should be `RPC API .....`

Gah! Thanks for catching.

Comment (Clay-Mysten) [reply doc/src/learn/index.md]: Fixed.

**Review by patrickkuo — APPROVED** [review]
Thanks! LGTM

**Pull request merged**